src/index.tsx

```diff
@@ -674,27 +674,37 @@ const StyledEditor = styled("div")<{
     position: relative;
   }
 
-  img {
-    max-width: 100%;
-  }
-
   .image {
     text-align: center;
-
+    max-width: 100%;
     img {
       pointer-events: ${props => (props.readOnly ? "initial" : "none")};
+      display: inline-block;
+      max-width: 100%;
+      max-height: 75vh;
     }
   }
 
   .image.placeholder {
     position: relative;
     background: ${props => props.theme.background};
-
     img {
       opacity: 0.5;
     }
   }
 
+  .image-half-right {
+    float: right;
+    width: 50%;
+    margin-left: 2em;
+  }
+
+  .image-half-left {
+    float: left;
+    width: 50%;
+    margin-right: 2em;
```

Comment:
After testing with some real content I think we should add `margin-bottom: 1em` too otherwise paragraphs wrap too close to the caption